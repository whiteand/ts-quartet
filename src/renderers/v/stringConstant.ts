import { ISchemaRenderer } from "../../types";
import { fromValueSchemaAlloc } from "./utils";

const MAX_LENGTH_STORED_IN_CONSTANT_LITERAL = 40;

export const stringConstantRenderer: ISchemaRenderer<
  string
> = fromValueSchemaAlloc(
  (valueId, schema, alloc) => {
    if (schema.length > MAX_LENGTH_STORED_IN_CONSTANT_LITERAL) {
      const stringVar = alloc(schema, "validString");
      return `${valueId} === ${stringVar}`;
    }
    return `${valueId} === ${JSON.stringify(schema)}`;
  },
  (valueId, schema, alloc) => {
    if (schema.length > MAX_LENGTH_STORED_IN_CONSTANT_LITERAL) {
      const stringVar = alloc(schema, "validString");
      return `${valueId} !== ${stringVar}`;
    }
    return `${valueId} !== ${JSON.stringify(schema)}`;
  }
);
