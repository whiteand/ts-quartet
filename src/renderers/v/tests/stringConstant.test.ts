import { stringConstantRenderer } from "../stringConstant";
import { snapshot } from "./snapshot";

describe("v constant number renderer", () => {
  test('v("Andrew")', () => {
    snapshot(stringConstantRenderer, { schema: "Andrew" });
  });
  test('v("Andrew is the creator of quartet data validation library. Try to create another long string")', () => {
    snapshot(stringConstantRenderer, {
      schema:
        "Andrew is the creator of quartet data validation library. Try to create another long string"
    });
  });
});
