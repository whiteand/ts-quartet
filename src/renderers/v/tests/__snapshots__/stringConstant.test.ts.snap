// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v constant number renderer v("Andrew is the creator of quartet data validation library. Try to create another long string") 1`] = `"_valueId === _validString_\\"Andrew is the creator of quartet data validation library. Try to create another long string\\""`;

exports[`v constant number renderer v("Andrew is the creator of quartet data validation library. Try to create another long string") 2`] = `"_valueId !== _validString_\\"Andrew is the creator of quartet data validation library. Try to create another long string\\""`;

exports[`v constant number renderer v("Andrew is the creator of quartet data validation library. Try to create another long string") 3`] = `"if (_valueId === _validString_\\"Andrew is the creator of quartet data validation library. Try to create another long string\\") return true;"`;

exports[`v constant number renderer v("Andrew is the creator of quartet data validation library. Try to create another long string") 4`] = `"if (_valueId !== _validString_\\"Andrew is the creator of quartet data validation library. Try to create another long string\\") return false;"`;

exports[`v constant number renderer v("Andrew") 1`] = `"_valueId === \\"Andrew\\""`;

exports[`v constant number renderer v("Andrew") 2`] = `"_valueId !== \\"Andrew\\""`;

exports[`v constant number renderer v("Andrew") 3`] = `"if (_valueId === \\"Andrew\\") return true;"`;

exports[`v constant number renderer v("Andrew") 4`] = `"if (_valueId !== \\"Andrew\\") return false;"`;
